tests/test_enterprise/api/test_views.py

```diff
@@ -7400,6 +7401,101 @@ def test_list_learner_statuses(self):
+        group = EnterpriseGroupFactory(
+            enterprise_customer=self.enterprise_customer,
+            applies_to_all_contexts=True,
+        )
+        pending_user = PendingEnterpriseCustomerUserFactory(
+            user_email="foobar@example.com",
+            enterprise_customer=self.enterprise_customer,
+        )
+        pending_user_query_string = f'?q={pending_user.user_email}'
+        url = settings.TEST_SERVER + reverse(
+            'enterprise-group-learners',
+            kwargs={'group_uuid': group.uuid},
+        ) + pending_user_query_string
+        response = self.client.get(url)
+
+        assert response.json().get('count') == 1
+        assert response.json().get('results')[0].get('pending_learner_id') == pending_user.id
+
+        group.applies_to_all_contexts = False
+
+        pending_membership = EnterpriseGroupMembershipFactory(
+            group=group,
+            pending_enterprise_customer_user=pending_user,
+            enterprise_customer_user=None,
+        )
+        existing_membership = EnterpriseGroupMembershipFactory(
+            group=group,
+            pending_enterprise_customer_user=None,
+            enterprise_customer_user__enterprise_customer=self.enterprise_customer,
+        )
+        existing_user_query_string = f'?q={existing_membership.enterprise_customer_user.user.first_name}'
+        url = settings.TEST_SERVER + reverse(
+            'enterprise-group-learners',
+            kwargs={'group_uuid': group.uuid},
+        ) + existing_user_query_string
+        response = self.client.get(url)
+
+        assert response.json().get('count') == 1
+        assert response.json().get('results')[0].get(
+            'learner_id'
+        ) == existing_membership.enterprise_customer_user.id
+
+        url = settings.TEST_SERVER + reverse(
+            'enterprise-group-learners',
+            kwargs={'group_uuid': group.uuid},
+        ) + pending_user_query_string
+
+        response = self.client.get(url)
+
+        assert response.json().get('count') == 1
+        assert response.json().get('results')[0].get(
+            'pending_learner_id'
+        ) == pending_membership.pending_enterprise_customer_user.id
+
+    def test_list_learners_sort_by(self):
+        """
+        Test that the list learners endpoint can be sorted by 'recentAction', 'memberStatus', 'memberDetails'
+        values respectively
+        """
+        # Test sorting by the three sortable values
+        for sort_by_value in ['recentAction', 'memberStatus', 'memberDetails']:
+            url = settings.TEST_SERVER + reverse(
+                'enterprise-group-learners',
+                kwargs={'group_uuid': self.group_1.uuid},
+            ) + f"?q={sort_by_value}"
+            response = self.client.get(url)
+            results = response.json().get('results')
+            returned_sorted_values = [
+                value.get(re.sub(r'(?<!^)(?=[A-Z])', '_', sort_by_value).lower()) for value in results
```

Comment:
There's already a helper method in this repo: https://github.com/openedx/edx-enterprise/blob/0e78b47c2d873c932e46826e031c357a5d0f9e76/enterprise/management/commands/manufacture_data.py#L56